src/app-interoperability/application-package-definition.linkml.yaml

```
@@ -0,0 +1,497 @@
+# yaml-language-server: $schema=https://linkml.io/linkml-model/linkml_model/jsonschema/meta.schema.json
+id: http://specification.margo.org/application-schema
+name: ApplicationDescription
+title: Application Description Schema
+description: >-
+  Schema for defining application metadata, deployment profiles, parameters, and configuration. 
+  The purpose of this file is to present the application on an application catalog or marketplace 
+  from where an end user selects the application to hand it over to the Workload Orchestration Software, 
+  which configures it and makes it available for installation on the edge device.
+version: 1.0 #Arne: update later
+prefixes:
+  linkml: https://w3id.org/linkml/
+  margo: https://specification.margo.org/
+imports:
+  - linkml:types
+
+default_prefix: margo
+
+
+# Class Definitions
+classes:
+  ApplicationDescription:
+    description: Root class for an application description.
+    attributes:
+      apiVersion:
+        description: Identifier of the version of the API the object definition follows.
+        required: true
+        range: string
+        rank: 10
+      kind:
+        description: Specifies the object type; must be 'application'.
+        range: string
+        required: true
+        pattern: ^application$ #Arne: should this be changed to "ApplicationDescription"?
+        rank: 20
+      metadata:
+        description: >-
+          Metadata element specifying characteristics about the application deployment.
+          See the [Metadata Attributes](#metadata-attributes) section below.
+        range: Metadata
+        required: true
+        rank: 30
+      deploymentProfiles:
+        description: >-
+          Deployment profiles element specifying the types of deployments the application supports. 
+          See the [Deployment](#deployment-profiles-attributes) section below.
+        range: DeploymentProfile #Arne: mistake in spec: called "Deployment" but type is called "DeploymentProfile"
+        multivalued: false
```

@arne-broering this attribute is being declared as a non-multivalued, but the examples appear to have arrays

```suggestion
        range: DeploymentProfile #Arne: mistake in spec: called "Deployment" but type is called "DeploymentProfile"
        multivalued: true
        inlined: true
        inlined_as_list: true
```